sphinx/source/docs/user_guide/extensions.rst

```diff
@@ -15,10 +15,14 @@ that let you:
 * Create highly specialized models for domain-specific use cases
 
 You can make and use custom extensions with standard releases and don't need to
-set up a development environment or build anything from source. This is the
-easiest way to get involved in Bokeh development. You can try new features and
-improved functionality without having to wait for the core team to implement
-them.
+set up a development environment or build anything from source. With custom
+extensions, you can also try new features and improved functionality without
+having to wait for the core team to implement them into Bokeh itself.
+
+.. note::
+    Extending Bokeh is a highly advanced feature. Several aspects of creating
+    and using extensions are still under development and should be considered
+    experimental.
```

Comment:
Is it worth enumerating things we *do* consider "stable"?

* existing models and properties
* `change.emit`, etc.

Reply:
Do you mean a list of stable elements of Bokeh in general or of stable elements in the context of developing and using extensions? 

I think a general list of what we consider stable/testing/experimental/etc in Bokeh would almost be too broad for the extensions chapter. If we want to start having a definitive list of which features we consider stable or experimental, something like the release notes, "Defining Key Concepts" in the user guide, and/or the developers guide might be a better place for that.

If you are talking about a list of stable elements to rely on specifically when developing and using extensions, I'd appreciate some more information on what to include in that list.

Reply:
I mean a list of things that are relatively safe to do in extensions. I've suggested a couple of things above, but we would need some input from @mattpap to make confident claims.

Reply:
We may want to make this a 3.0 task, and formally define the public facing APIs (beyond the trivia like modes and properties).

Reply:
OK I am fine to wait. My only other comment @tcmetzger is a few wording suggestions